helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/CuratorClientFactory.java

```diff
@@ -18,14 +18,19 @@
 package com.spotify.helios.servicescommon.coordination;
 
 import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.AuthInfo;
 import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.api.ACLProvider;
+
+import java.util.List;
 
 public interface CuratorClientFactory {
 
-    public CuratorFramework newClient(String connectString,
-                                      int sessionTimeoutMs,
-                                      int connectionTimeoutMs,
-                                      RetryPolicy retryPolicy,
-                                      String namespace);
+  public CuratorFramework newClient(String connectString,
+                                    int sessionTimeoutMs,
+                                    int connectionTimeoutMs,
+                                    RetryPolicy retryPolicy,
+                                    ACLProvider aclProvider,
```

Comment:
Why? We'd still have to check if the optional is null and then check if it contains a value. Right now we just check if the object is null.